api/groups_test.go

```diff
@@ -895,3 +898,241 @@ func TestCreateNewGroup(t *testing.T) {
+				Description:      aws.String("A test group1"),
+				GroupName:        aws.String("test-group1"),
+				LastModifiedDate: &timestamp,
+				Precedence:       aws.Int64(4),
+				RoleArn:          aws.String(""),
+				UserPoolId:       aws.String(""),
+			},
+		}
+	)
+
+	api, w, m := apiSetup()
+
+	Convey("List groups -check expected responses", t, func() {
+		internalErrorDescription := ""
+		listGroupsTest := []struct {
+			description           string
+			next_token            string
+			getListGroupsFunction func(input *cognitoidentityprovider.ListGroupsInput) (*cognitoidentityprovider.ListGroupsOutput, error)
+			assertions            func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse)
+		}{
+			{
+				"200 response from Cognito with empty NextToken",
+				"",
+				func(input *cognitoidentityprovider.ListGroupsInput) (*cognitoidentityprovider.ListGroupsOutput, error) {
+					return &cognitoidentityprovider.ListGroupsOutput{
+						Groups:    groups,
+						NextToken: nil,
+					}, nil
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+
+					So(errorResponse, ShouldBeNil)
+
+					So(successResponse.Status, ShouldEqual, http.StatusOK)
+					So(successResponse, ShouldNotBeNil)
+					So(successResponse.Body, ShouldNotBeNil)
+
+					var responseBody = models.ListUserGroups{}
+					json.Unmarshal(successResponse.Body, &responseBody)
+
+					So(responseBody.NextToken, ShouldBeNil)
+					So(responseBody.Count, ShouldEqual, 2)
+					So(responseBody.Groups, ShouldNotBeNil)
+					So(responseBody.Groups, ShouldHaveLength, responseBody.Count)
+					So(*responseBody.Groups[0].Description, ShouldEqual, *groups[0].Description)
+				},
+			},
+			{
+				"200 response from Cognito with no groups",
+				"",
+				func(input *cognitoidentityprovider.ListGroupsInput) (*cognitoidentityprovider.ListGroupsOutput, error) {
+					return &cognitoidentityprovider.ListGroupsOutput{
+						Groups:    []*cognitoidentityprovider.GroupType{},
+						NextToken: nil,
+					}, nil
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+
+					So(errorResponse, ShouldBeNil)
+
+					So(successResponse.Status, ShouldEqual, http.StatusOK)
+					So(successResponse, ShouldNotBeNil)
+					So(successResponse.Body, ShouldNotBeNil)
+
+					var responseBody = models.ListUserGroups{}
+					json.Unmarshal(successResponse.Body, &responseBody)
+					So(responseBody.NextToken, ShouldBeNil)
+					So(responseBody.Count, ShouldEqual, 0)
+					So(responseBody.Groups, ShouldBeNil)
+					So(responseBody.Groups, ShouldHaveLength, responseBody.Count)
+				},
+			},
+			{
+				"200 response from Cognito with populated NextToken",
+				"next_token",
+				func(input *cognitoidentityprovider.ListGroupsInput) (*cognitoidentityprovider.ListGroupsOutput, error) {
+					return &cognitoidentityprovider.ListGroupsOutput{
+						Groups:    groups,
+						NextToken: nil,
```

Comment:
I think this should be a non nil value for this test case?

Reply:
return NextToken will always be empty  but is used in the input   from the pagination loop will have no next_token 
![awspagination](https://user-images.githubusercontent.com/81749650/135494750-298a30f5-3aa1-4b8b-aa3b-0045e68c97e6.png)

Reply:
so the tests are faking an input with token....